library/core/class.session.php

```diff
@@ -656,6 +659,10 @@ public function stash($name = '', $value = '', $unsetOnRetrieve = true) {
             return;
         }
 
+        // Create a fresh copy of the Sql object to avoid pollution.
+        $this->sql = clone Gdn::sql();
+        $this->sql->reset();
```

Comment:
These 2 lines should be done only one time.
Cloning an object on every `stash()`call is not optimal.